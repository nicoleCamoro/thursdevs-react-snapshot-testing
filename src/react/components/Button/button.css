.dog-btn {
    flex-shrink: 0;
    font-family: Arial, sans-serif;
    font-weight: bolder;
    font-size: 20px;
    border: 2px solid #27272d;
    border-radius: 10px;
    margin-top: calc(15px + 2vh);
    padding: 15px 20px;
    background: none;
    color: #131316;
    background-color: #d1d4eb;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
    outline: none;
    cursor: pointer;
    transition: all 200ms;
}

.dog-btn:active {
    transform: translateY(2px);
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.16), 0 1px 2px rgba(0, 0, 0, 0.23);
}
