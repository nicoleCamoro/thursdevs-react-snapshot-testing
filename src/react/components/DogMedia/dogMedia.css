.dog-media {
    width: 100%;
    height: calc(100% - 98px);
    object-fit: cover;
    border: 1px solid #eee;
}
