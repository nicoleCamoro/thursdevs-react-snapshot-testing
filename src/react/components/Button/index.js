import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './button.css';

class Button extends Component {
    render() {
        return (
            <button className="dog-btn" onClick={this.props.onClick}>
                Get Random Doggo
            </button>
        );
    }
}

Button.propTypes = {
    onClick: PropTypes.func.isRequired,
};

export default Button;
