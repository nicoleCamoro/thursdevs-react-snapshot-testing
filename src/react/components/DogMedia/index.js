import React, { Component } from 'react';
import './dogMedia.css';

class DogMedia extends Component {
    render() {
        return (
            <div>
                {this.props.mediaType === 'img' ? (
                    <img src={this.props.dogURL} className="dog-media" />
                ) : (
                    <video
                        src={this.props.dogURL}
                        autoplay=""
                        muted=""
                        loop=""
                        className="dog-media"
                    />
                )}
            </div>
        );
    }
}

export default DogMedia;
