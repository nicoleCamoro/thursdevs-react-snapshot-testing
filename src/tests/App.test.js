import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../react/components/Button';
import DogMedia from '../react/components/DogMedia';

// describe('Button component', () => {
//     it('Button components renders initially', () => {
//         const component = renderer.create(<Button onClick={() => {}} />);

//         expect(component.toJSON()).toMatchSnapshot();
//     });

//     it('renders with null as an initial state', () => {
//         const component = renderer.create(<Button onClick={() => {}} />);
//         const instance = component.getInstance();
//         // console.log(instance);

//         expect(instance.state).toBeNull();
//     });
// });

// describe('DogMedia component', () => {
//     it('renders video', () => {
//         const component = renderer.create(<DogMedia mediaType="video" />);
//         const instance = component.getInstance();

//         expect(instance.props.mediaType).toBe('video');
//     });
// });

// function sum(a, b) {
//     return a + b;
// }

// test('adds 1 + 2 to equal 3', () => {
//     expect(sum(1, 2)).toBe(3);
// });
