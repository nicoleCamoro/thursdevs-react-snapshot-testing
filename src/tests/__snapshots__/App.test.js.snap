// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button components renders initially 1`] = `
<button
  className="dog-btn"
  onClick={[Function]}
>
  Get Random Doggo
</button>
`;
